### PR TITLE
Milestone 1: OOP school library (refactor your code)

### PR DESCRIPTION
**Pull request by Gardimy**
**In this project, the main focus is on refactoring the existing codebase to improve its maintainability, readability, and adherence to best practices. I analyze my previous code and:** 
- Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Pay special attention to the "Single-responsibility Principle".
. Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
. Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@
```
Comment (V-Blaze) [inline README.md:21]: - You have done great so far :+1: :+1: But there are some requirements you still need to meet before moving to the next milestone. :+1: 
- [ ] As the aim of this milestone is to follow the best practices for KISS, DRY code, I'll suggest you move the choice selection to your `main.rb`. `main.rb` is only to start their code and ask the user for the initial choice of options (or to quit).

- - Screenshot:
Screenshot from 2023-08-14 23-29-34

File: README.md [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 # 📖OOP school library <a name="about-project"></a>
 
-**OOP school library** In this project, I create a form of UI for your school library app. This way it can be invoked as an executable and not something you use in IRB exclusively.
+**OOP school library** In this project, the main focus is on refactoring the existing codebase to improve its maintainability, readability, and adherence to best practices.
 
 ## 🛠 Built With <a name="built-with"></a>
```
Comment (V-Blaze) [inline README.md:26]: - [ ] Following the ``Single responsibility rule``, the validation and checking of the persons type should be placed on a different method, which should in turn call the `create_student` or `create_teacher` method depending on the option selected, I'll suggest you update your code to follow this guideline. :+1: :+1: :100: 

- - Code section screenshot:
Selection_003

- [ ] Also while creating a person, you didn't provide a prompt to take the person's name, I'll suggest you update that also as all the person tends to coemup with the default name of `unknown`. :+1: :+1: :small_airplane:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Gardimy 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Correct GIt flow✔
- Meaningful commit messages👋 
-  Readme looks professional ✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by addod19 — APPROVED** [review]
## Hi @Gardimy, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Nice job addressing all the comments by the previous code reviewer :rocket: 

### Optional
- [ ] This project is a pair programming activity, since the previous reviewer did not point it out in the first review or invalidate it, for consistency I'm reviewing it. Going forward, let us try to follow requirements, it is an important part of our programming journey
- [ ] It would be nice if we fix the bug, creating a rental stores the data double as we can see in the screenshot below
Screenshot from 2023-08-15 03-13-25



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**